### PR TITLE
Test specializations

### PR DESCRIPTION
**Pull request by phimuemue**
We have more and more specializations, but so far they are not vastly tested for iterators that have already been advanced. This PR alters `test_specializations`.

### CODE REVIEW

**Comment by Philippe-Cholet** [comment]
I worked some time ago on a (maybe too complex) way to test specializations (draft: https://github.com/rust-itertools/itertools/issues/756#issuecomment-1739905965) before I split focus on other things.
Here, 5 as limit might be too little, at least fior some iterators, I would suggest up to 10.

**Comment by phimuemue** [comment]
@Philippe-Cholet Can you merge this? I‘ll raise the limit to ten when I have time.

Side note: I just filled this as a PR to have the CI/Tests/Checks running, but it seems, I cannot merge myself anymore (as it was the case back then with bors). Am I doing something wrong?

**Review by Philippe-Cholet — APPROVED** [review]

**Comment by Philippe-Cholet** [comment]
All the maintainer part is new to me but I guess we can not merge a PR (except jswrenn I suppose as he is an owner) especially if it has a commit we wrote without explicit approval, something like that. I think it would have work if you approved your own changes.

**Pull request merged**
